doc/development/development_guide.rst

```diff
@@ -127,24 +127,65 @@ Individual test modules are run by invoking pytest directly from the command lin
 
 .. note:: **Adding tests to Strawberry Fields**
 
-    The ``tests`` folder is organised into four subfolders: ``backend`` for tests that
-    only import a Strawberry Fields backend, ``frontend`` for tests that import the Strawberry
-    Fields UI but do not make use of a backend, ``integration`` for tests that test
-    integration of the frontend and backends, and ``apps`` for tests of the applications layer.
+    The ``tests`` folder is organised into several subfolders:
 
-    When writing new tests, make sure to mark what components it tests. For a backend test,
-    you can use the ``backends`` mark, which accepts the names of the backends:
+    - ``backend`` for tests that only import a Strawberry Fields backend,
+    - ``frontend`` for tests that import the Strawberry Fields UI but do not make use of a backend,
+    - ``integration`` for tests that test integration of the frontend and backends,
+    - ``apps`` for tests of the applications layer
+
+    When writing new tests, make sure to mark what components it tests.
+
+    Certain tests that are related to a specific backend, e.g. test cases for
+    its operations or the states returned by a backend. For a backend test, you
+    can use the ``backends`` mark, which accepts the names of the backends:
 
     .. code-block:: python
 
         pytest.mark.backends("fock", "gaussian")
 
-    For a frontend-only test, you can use the frontend mark:
+    So a specific test case could be decorated as follows:
+
+    .. code-block:: python
+
+        @pytest.mark.backends("fock", "gaussian")
+        def test_fock_and_gaussian_feature():
+
+    Adding tests for an engine, operations, parameters and other parts of the
+    user interface can be added as part of the frontend tests. For a
+    frontend-only test, you can use the frontend mark:
 
     .. code-block:: python
 
         pytest.mark.frontend
```

Comment:
It might be good to explain why this doesn't have a decorator. That is, you can use it on the module level:

```python
mark = pytest.mark.frontend
```

which will mark the entire test file as a frontend test. The decorator is instead used to mark only classes and test functions

Reply:
Thanks, extended